From c729f8602bc51ce463d463aa0f5435b551ab0e2e Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Sun, 29 Dec 2019 19:56:21 -0300
Subject: [PATCH 2/2] Fix speech for Farsi and Hindi

These two languages dictionaries do not follow
the same structure as the other ones, so treat
them as special cases.
---
 wordsactivity.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/wordsactivity.py b/wordsactivity.py
index 38093f6..a1d8396 100644
--- a/wordsactivity.py
+++ b/wordsactivity.py
@@ -583,8 +583,14 @@ class WordsActivity(SugarCompatibleActivity):
         # remove the lines with the english definition
         clean_text = ''
         logging.debug('text %s', text)
-        for line in text.split('\n'):
-            if len(line) > 0 and line[0] in (' ', '\t'):
+        for index, line in enumerate(text.split('\n')):
+            if self.destination_lang not in ['hin', 'ara'] and len(
+                    line) > 0 and line[0] in (' ', '\t'):
+                clean_text += line + ','
+            if self.destination_lang == 'hin' and re.search(
+                    '^(\d*\.)', line, 0):
+                clean_text += re.sub('^(\d*\.)', '', line) + ','
+            if self.destination_lang == 'ara' and index != 0 and len(line) > 0:
                 clean_text += line + ','
         # remove text between []
         clean_text = re.sub('\[.*?\]', '', clean_text)
-- 
2.23.0

